Defaulting to latest

gambol99 opened the pull request:
## Summary

The issue with having this set to the latest tag is those build of master SHOULD get master/latest else you can end up with weird edge cases due to the fact the tag is carried through to things like the images versions and the github release. This change should make it work as as expected i.e. if you build the binaries off master you get latest images and if you pull from a release it's tagged with an actual release version. 

Issue was hit by some people whom were trying out Kore, building binaries off master and hitting issues.

**Which issue(s) this PR resolves**:

Resolves BUG

## Checklist

 - [ ] I've created a PR to update the [documentation](https://github.com/appvia/kore-docs): NONE

## Changes

When building from master you get latest images. CircleCI passes the tag into the build and so binaries built from a tag trigger will have the version embedded.

## Follow-up stories

NA

bandesz commented:
I don't think this is a good solution, and even the current state is bad, because if someone ever runs a docker image build make target without setting VERSION, you can easily override the last release tag or latest from your branch.

I propose a new solution, where by default version is "dev-<CURRENT GIT REF>" (optionally with "-dirty" added which doesn't matter).
 * if the version starts with "dev-", we use "latest" by for all Docker images
 * otherwise we use the same version value as Docker image tags.

bandesz commented on hack/verify-release-images.sh:
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
```
set -o errexit
set -o nounset
set -o pipefail
```

gambol99 replied:
28090288bea59cb0968936d2262f04bcace2edce

bandesz commented on hack/verify-release-images.sh:
```diff
@@ -0,0 +1,43 @@
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+FORCE_IMAGE_PUSH=${FORCE_IMAGE_PUSH:-false}
+VERSION=${VERSION:-latest}
+PROTECT="master"
```
This should be `PROTECT="latest"

gambol99 replied:
28090288bea59cb0968936d2262f04bcace2edce

bandesz commented on hack/verify-release-images.sh:
```diff
@@ -0,0 +1,43 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
```
This will still allow me to push local changes, if I'm on the master branch and I built images with my local changes. I still think using `latest-[ref]` is the safest choice as we might forget to put verify-release-images.sh to all places in the future. That way you will have a proper Git revision in your local builds too.

Although it's a separate problem, but VERSION defaults to "latest" in the UI, so we should also have the same protection there.

gambol99 replied:
28090288bea59cb0968936d2262f04bcace2edce

gambol99 commented:
> This will still allow me to push local changes, if I'm on the master branch and I built images with my local changes. I still think using latest-[ref] is the safest choice as we might forget to call verify-release-images.sh in every required places in the future. That way you will have a proper Git revision in your local builds too.

I've fixed up the check and added to the ui as well to make it both consistent. In regard to `latest-ref` ... we could just use the SHA and merge to master we always push latest and image:sha ...It might get annoying though for local dev .. i.e. i on occasion build image with change and kind load docker image into the cluster but now i'd have to change the deployment as well as the image has changed ... Or testing changes out remotely .. the auth proxy, i usually push over the image and reload with pull image always.

bandesz commented:
@gambol99 I don't understand. Anyway, I will approve this, as I don't think I can convince you.
